### PR TITLE
Revert tests for multi-controlled-zyz

### PR DESCRIPTION
**Pull request by maliasadi**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
PR https://github.com/PennyLaneAI/pennylane/pull/6042 adds support for multi-controlled zyz.
For Lightning, this means that custom decompositions to special zyz with multiple controlled wires will be supported.
 
**Description of the Change:**
- Add tests for adjoint jacobian of the controlled gate with multiple controlled_wires. 

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-67711]

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 pip~=22.0
-git+https://github.com/PennyLaneAI/pennylane.git@master
+git+https://github.com/PennyLaneAI/pennylane.git@multicontrolled-zyz
```
Comment (maliasadi) [inline requirements-dev.txt:2]:
```suggestion
git+https://github.com/PennyLaneAI/pennylane.git@master
```

File: tests/test_adjoint_jacobian.py [hunk 2]
```diff
@@ -865,13 +865,12 @@ def test_controlled_jacobian(self, par, n_qubits, control_value, operation, tol)
...
-                    # decomposition to `ctrl_decomp_zyz` works now with only
-                    # one single controlled wire.
                     qml.ctrl(
                         operation(*p, wires=range(n_qubits - num_wires, n_qubits)),
-                        control_wires[0],
-                        control_values=control_value,
+                        control_wires,
+                        control_values=[
+                            control_value or bool(i % 2) for i, _ in enumerate(control_wires)
+                        ],
```
Comment (LuisAlfredoNu) [inline tests/test_adjoint_jacobian.py:873]: Nice way to initialize the control values :rocket:

**Review by LuisAlfredoNu — APPROVED** [review]
LGTM. Thank you @maliasadi  :+1:

**Comment by multiphaseCFD** [comment]
Looks good to me! Thanks @maliasadi !

**Review by multiphaseCFD — COMMENTED** [review]
Thanks @maliasadi ! Looks good to me!

**Review by multiphaseCFD — APPROVED** [review]

**Pull request merged**